v3/newrelic/config.go

```diff
@@ -397,29 +411,102 @@ type Config struct {
-// means to include code level metrics everywhere currently supported.
+// attach code level metric data.
+//
+// Currently, this includes
+//    TransactionCLM            any kind of transaction
+//    AllCLM                    all kinds of telemetry data for which CLM is implemented (the default)
 //
-// A scope of 0 means "all types" as a convenience so that any zero-value
-// CodeLevelMetricsScope variable provides the default expected behavior
-// rather than turning off all code level metrics.
+// The zero value of CodeLevelMetricsScope means "all types" as a convenience so that
+// new variables of this type provide the default expected behavior
+// rather than, say, turning off all code level metrics as a 0 bit value would otherwise imply.
+// Otherwise the numeric values of these constants are not to be relied
+// upon and are subject to change. Only use the named constant identifiers in
+// your code. We do not recommend saving the raw numeric value of these constants
+// to use later.
 const (
-	TransactionCLM CodeLevelMetricsScope = 1 << iota // include CLM data in transactions
-	AllCLM         CodeLevelMetricsScope = 0         // all supported types
+	TransactionCLM CodeLevelMetricsScope = 1 << iota
+	AllCLM         CodeLevelMetricsScope = 0
 )
 
-func codeLevelMetricsScopeLabelToValue(label string) (CodeLevelMetricsScope, bool) {
-	switch label {
-	case "all":
-		return AllCLM, true
+//
+// CodeLevelMetricsScopeLabelToValue accepts a number of string values representing
+// the possible scope restrictions available for the agent, returning the
+// CodeLevelMetricsScope value which represents the combination of all of the given
+// labels. This value is suitable to be presented to ConfigCodeLevelMetricsScope.
+//
+// It also returns a boolean flag; if true, it was able to understand all of the
+// provided labels; otherwise, one or more of the values were not recognized and
+// thus the returned CodeLevelMetricsScope value may be incomplete (although it
+// will represent any valid label strings passed, if any).
+//
+// Currently, this function recognizes the following labels:
+//   for AllCLM: "all" (if this value appears anywhere in the list of strings, AllCLM will be returned)
+//   for TransactionCLM: "transaction", "transactions", "txn"
+//
+func CodeLevelMetricsScopeLabelToValue(labels ...string) (CodeLevelMetricsScope, bool) {
+	var scope CodeLevelMetricsScope
+	ok := true
+
+	for _, label := range labels {
+		switch label {
+		case "":
+
+		case "all":
+			return AllCLM, true
+
+		case "transaction", "transactions", "txn":
+			scope |= TransactionCLM
 
-	case "transaction", "transactions", "txn":
-		return TransactionCLM, true
+		default:
+			ok = false
+		}
 	}
-	return 0, false
+	return scope, ok
+}
+
+//
+// UnmarshalJSON allows for a CodeLevelMetricsScope value to be read from a JSON
+// string whose value is a comma-separated list of scope labels.
+//
+func (s *CodeLevelMetricsScope) UnmarshalJSON(b []byte) error {
```

Comment:
Please implement `UmarshalText` instead as it is more generic. (It will be used by JSON, XML, and others.)

Reply:
That was trivial enough that I just pushed the change for it. I'll look at the other suggestions next.